File: Backend-SpringBoot/FoodApp/src/main/java/com/tientien/foodapp/menuitems/entity/MenuItem.java
package com.tientien.foodapp.menuitems.entity;

import com.tientien.foodapp.category.entity.Category;
import com.tientien.foodapp.common.entity.BaseEntity;
import com.tientien.foodapp.common.enums.MenuItemStatus;
import com.tientien.foodapp.restaurant.entity.Restaurant;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "menu_items")
@Getter
@Setter
public class MenuItem extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Double price;

    private Double discountPrice;

    @Enumerated(EnumType.STRING)
    private MenuItemStatus status;

    private String image;

    private String description;

    @ManyToOne
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}

File: Frontend_Flutter/lib/models/menu_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'menu_item.freezed.dart';
part 'menu_item.g.dart';

@freezed
class MenuItem with _$MenuItem {
  const factory MenuItem({
    int? id,
    required String name,
    required double price,
    double? discountPrice,
    String? status,
    String? image,
    String? description,
    int? restaurantId,
    String? restaurantName,
    int? categoryId,
    String? categoryName,
  }) = _MenuItem;

  factory MenuItem.fromJson(Map<String, dynamic> json) => _$MenuItemFromJson(json);
}
